instrumentation/jdbc/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jdbc/JdbcUtils.java

```diff
@@ -69,8 +72,27 @@ public static Connection connectionFromStatement(Statement statement) {
     return connection;
   }
 
+  public static SqlStatementInfo sanitizeAndExtractInfo(String sql) {
+    try {
+      return sqlToStatementInfoCache.get(
```

Comment:
Is it possible / makes sense to only use the cache with prepared statements? The bounding makes it safe, but I guess for raw queries the cache will just thrash all the time.

Reply:
There's another code path that uses `JdbcMaps.preparedStatements` for caching prepared statements.   It's a contract between the `JdbcTracer` and the `ConnectionInstrumentation`.  Would prefer to circle back on this later if it becomes a problem...although it might be difficult to trace without internal metrics.  I'll open an issue to track the idea of adding metrics for cache misses.